### ISSUE TITLE
Remove frequency data from indexes

### ISSUE THREAD

**Issue by miccolis** [mapbox/carmen#853]
**tl;dr** - we should remove frequency information from our indexes & plan to bring it back in the future as part of a data structure fuzzy-phrase can use.

## How do we calculate it now? What does it do?

When indexing documents we batch them into sets of 10,000 and calculate word frequency. Words that are appear a lot [insert definition] within this set of 10k documents, are deemed “low information". Features that contain that contain these “low information” words have versions of their text indexed at lower relevance with those words omitted. Assuming a uniform type of feature would work at great identifying common words (ie ‘street’, ‘way’) and indexing versions of features with those words omitted.

Unfortunately out data isn’t uniform. Our data is such that local history & geography can have a lot of influence on when a word is common or not. “Northeast”, “Northwest”, “Southeast”, “Southwest” are probably the most common words in the DC street name data set. They literally apply to every street. However, they are completely un-used across the river in Virginia. We have no control over how we batch records into 10k sets. Maybe we’ll do enough of DC in one go to indentify these as common, maybe we won’t.

Also, we don’t pay attention to language when we generate frequency information, which is just silly.

## Why eliminating it is ok?

We don’t actually use this frequency information at query time. Removing it from out indexes will have no negative affects.

## Where we’ll want to bring it back

We should use frequency information at query time and there are two lines of thinking here. Likely they’ll both prove useful and we’d implement both sequentially;

1. Given a list of low information words we could evaluate a users’s query with those words omitted. For example if “street” & “avenue” were deemed low information (very likely) we’d have a mechanism for handling when a user enters the wrong street type. To get this functionality we’d want to expose a datastructure to fuzzy-phrase that can make a determination about relevance when evaluating whether a index contains particular text.
2. With the above in place it may be possible to offload the work of dropping low information words from feature names entirely to query time. This could significantly reduce index size.

### Next Steps

* Actually do the removal. Will likely require changes in any dependent project. 

@mapbox/search

**Comment by miccolis** [comment]
Done in #854
